File: booksamples/chapter-05/src/main/java/org/vulkanb/Main.java
package org.vulkanb;

import org.tinylog.Logger;
import org.vulkanb.eng.*;
import org.vulkanb.eng.graph.Render;
import org.vulkanb.eng.scene.Scene;

public class Main implements IAppLogic {

    public static void main(String[] args) {
        Logger.info("Starting application");

        Engine engine = new Engine("Vulkan Book", new Main());
        engine.start();
    }

    @Override
    public void cleanup() {
        // To be implemented
    }

    @Override
    public void init(Window window, Scene scene, Render render) {
        // To be implemented
    }

    @Override
    public void input(Window window, Scene scene, long diffTimeMillis) {
        // To be implemented
    }

    @Override
    public void update(Window window, Scene scene, long diffTimeMillis) {
        // To be implemented
    }
}

File: booksamples/chapter-05/kotlin/src/main/kotlin/org/vulkanb/Main.kt
package org.vulkanb

import io.github.oshai.kotlinlogging.KotlinLogging
import org.vulkanb.eng.Engine
import org.vulkanb.eng.IAppLogic
import org.vulkanb.eng.Window
import org.vulkanb.eng.graph.Render
import org.vulkanb.eng.scene.Scene

private val logger = KotlinLogging.logger {}

class Main : IAppLogic {
    override fun cleanup() {
        // To be implemented
    }

    override fun init(window: Window?, scene: Scene?, render: Render?) {
        // To be implemented
    }

    override fun input(window: Window?, scene: Scene?, diffTimeMillis: Long) {
        // To be implemented
    }

    override fun update(window: Window?, scene: Scene?, diffTimeMillis: Long) {
        // To be implemented
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            logger.info { "Starting application" }
            val engine = Engine("Vulkan Book", Main())
            engine.start()
        }
    }
}
